Consolidate reusable parameters and reusable objects. (#182)
OAI/sig-workflows: versions/1.0.0.md

}
```

#### Reusable Object

A simple object to allow referencing of objects contained within the [Components Object](#components-object). It can be leveraged from locations within steps or workflows in the Workflows Description. **Note** - Input Objects MUST use standard JSON Schema referencing via the `$ref` keyword while all non JSON Schema objects will leverage this object and it's expression based referencing mechanism.

##### Fixed Fields
Field Name | Type | Description
---|:---:|---
<a name="reusableObjectName"></a>reference | `{expression}` | **REQUIRED**. A [runtime expression](#runtime-expressions) used to reference the desired object.

> <a name="reusableObjectValue"></a>value | `string` | A value which by default MUST override that of the referenced parameter. This is only applicable for parameter object references.

Reviewer: What does it mean "by default"? Are those words necessary? I wonder why this only applies to parameter object references?
```suggestion
<a name="reusableObjectValue"></a>value | `string` | Sets a value of the referenced parameter. This is only applicable for parameter object references.

```

This object cannot be extended with additional properties and any properties added MUST be ignored.

##### Reusable Object Example

```yaml
  reference: $components.successActions.notify
```

```json
  {
    "reference": "{$components.successActions.notify}"
  }
```

```yaml
  reference: $components.parameters.page
  value: 1
```

```json
  {
    "reference": "{$components.parameters.page}",

>     "value": 1

Reviewer 2: Suggest we make these consistent - JSON and YAML examples - this implies to me that JSON must have a `{` which is not true. (Same with the example above)
